tests/UI/pages/BO/catalog/discounts/add.js

```diff
@@ -10,55 +10,192 @@ class AddCartRule extends BOBasePage {
     await page.click(this.conditionsTabLink);
-    await this.setValue(page, this.singleCustomerInput, cartRuleData.customer);
-    await this.waitForVisibleSelector(page, `${this.singleCustomerResultBlock}:not([style*='display: none;'])`);
-    await page.click(this.singleCustomerResultItem);
-    // Fill actions form
+
+    // Set Customer
+    // Customer will not be set if we want to use the cart rule for any customer
+    if (cartRuleData.customer) {
+      await this.setValue(page, this.singleCustomerInput, cartRuleData.customer);
+      await this.waitForVisibleSelector(page, `${this.singleCustomerResultBlock}:not([style*='display: none;'])`);
+      await page.click(this.singleCustomerResultItem);
+    }
+
+    // Fill date from if its changed
+    if (cartRuleData.dateFrom) {
+      await this.setValue(page, this.dateFromInput, cartRuleData.dateFrom);
+    }
+
+    // Fill date to if its changed
+    if (cartRuleData.dateTo) {
+      await this.setValue(page, this.dateToInput, cartRuleData.dateTo);
+    }
+
+    // Fill minimum amount values
+    await this.setValue(page, this.minimumAmountInput, cartRuleData.minimumAmount.value.toString());
+    await this.selectByVisibleText(page, this.minimumAmountCurrencySelect, cartRuleData.minimumAmount.currency);
+    await this.selectByVisibleText(page, this.minimumAmountTaxSelect, cartRuleData.minimumAmount.tax);
+    await this.selectByVisibleText(page, this.minimumAmountShippingSelect, cartRuleData.minimumAmount.shipping);
+
+    // Fill quantities
+    await this.setValue(page, this.quantityInput, cartRuleData.quantity.toString());
+    await this.setValue(page, this.quantityPerUserInput, cartRuleData.quantityPerUser.toString());
+  }
+
+
+  /**
+   * Fill actions tab
+   * @param page
+   * @param cartRuleData
+   * @return {Promise<void>}
+   */
+  async fillActionsForm(page, cartRuleData) {
+    // Go to actions tab
     await page.click(this.actionsTabLink);
+
+    // Set free shipping toggle
     await page.click(this.freeShippingInput(cartRuleData.freeShipping));
 
-    if (cartRuleData.percent) {
-      await page.click(this.applyDiscountRadioButton('percent'));
+    switch (cartRuleData.discountType) {
+      case 'Percent':
+        await page.click(this.discountPercentRadioButton);
+        await this.setValue(page, this.discountPercentInput, cartRuleData.discountPercent.toString());
+        break;
+      case 'Amount':
+        await page.click(this.discountAmountRadioButton);
+        await this.setValue(page, this.discountAmountInput, cartRuleData.discountAmount.value.toString());
+        await this.selectByVisibleText(page, this.discountAmountCurrencySelect, cartRuleData.discountAmount.currency);
+        await this.selectByVisibleText(page, this.discountAmountTaxSelect, cartRuleData.discountAmount.tax);
+        break;
+      case 'None':
+        await page.click(this.discountOffRadioButton);
+        break;
+      default:
+        // Do nothing for this option
```

Comment:
Could be for the `'None'` option then, no ? Or throwing an error since you're not supposed to finish here ?

Reply:
It's better to throw an error

Reply:
done thanks